pkg/appmanager/appManager.go

```diff
@@ -3141,14 +3141,18 @@ func (appMgr *Manager) getEndpointsForNodePort(
 ) []Member {
 	nodes := appMgr.getNodesFromCache()
 	var members []Member
+	uniqueMembersMap := make(map[Member]bool)
 	for _, v := range nodes {
 		member := Member{
 			Address: v.Addr,
 			Port:    nodePort,
 			SvcPort: port,
 			Session: "user-enabled",
 		}
-		members = append(members, member)
+		if _, value := uniqueMembersMap[member]; !value {
```

Comment:
minor comments - we could use empty struct for value type, rename the variable "value" to "found/exists"